### ISSUE TITLE
Make LDAP Sync Better to configure

### ISSUE THREAD

**Issue by grisuthedragon** [csf-ngs/hinkskalle#34]
Since I got hinkskalle running and the first experiments were statisfying I started integrating it in our infrastructure. Thereby, the LDAP sync allows us to login with our normal usersnames (something like lastname+x), hinkskalle uses `lowercase(givenname.lastname)`, which is not intuitive in our setup. The reason for this is the following line:

https://github.com/csf-ngs/hinkskalle/blob/e25a5dca577dba95558437f5739d69781844afc6/backend/Hinkskalle/util/auth/ldap.py#L99

in my installation, I changed it to:
```
user.username = _get_attr(attrs.get('uid'))
```

Since everybody has a bit different ldap installation, this should be configurable.

**Comment by h3kker** [comment]
That's a very good point. In Hinkskalle's original ecosystem uid == cn. I took a quick peek at some ldap specs and it seems that this is not the usual case. [`cn`](https://ldapwiki.com/wiki/Cn) is typically the full name, while [`uid`](https://ldapwiki.com/wiki/Uid) corresponds to the short, login id. 

I think I should change the default to `uid`, but a configuration option would be good, I agree.

**Comment by grisuthedragon** [comment]
In our environment, we have two `cn` entries, but the first one is the full name and the second equals the uid. And thus it is hard to choose which one is the correct one for the use case.  In general, I am a friend of having configurable solutions but with a good educated guess as default.

**Comment by h3kker** [comment]
I'm preparing 4.4.0 with flexible LDAP configuration. 

I would appreciate if you could check out [the updated docs](https://github.com/csf-ngs/hinkskalle/blob/ldap-config-attr-mapping/docs/configuration.md#fine-tuning) and tell me if that sounds reasonable. I also changed the logic a little bit, instead of implicitely enabling LDAP when a HOST was configured, now you have to set a ENABLED (either in the config.json or via `HINKSKALLE_LDAP_ENABLED=1` env). I think this is more obvious, and one can also switch on/off temporarily without messing with the host config. 

If you like you can also try it out, there's a container with the latest version from the branch/PR: `ghcr.io/csf-ngs/hinkskalle:ldap-config-attr-mapping`.

**Comment by grisuthedragon** [comment]
I took a look on the documentation and the only thing, which is somehow missing, is: How to deal with the `secrets.json` file mentioned for the LDAP bind password or do you means the `secrets.env` file? 

But the sync seems to be somehow damaged.

In the webfrontend I get:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/rq/worker.py", line 1061, in perform_job
    rv = job.perform()
  File "/usr/local/lib/python3.8/dist-packages/flask_rq2/job.py", line 41, in perform
    return super(FlaskJob, self).perform()
  File "/usr/local/lib/python3.8/dist-packages/rq/job.py", line 821, in perform
    self._result = self._execute()
  File "/usr/local/lib/python3.8/dist-packages/rq/job.py", line 844, in _execute
    result = self.func(*self.args, **self.kwargs)
  File "/srv/hinkskalle/backend/Hinkskalle/util/jobs.py", line 167, in sync_ldap
    svc.ldap.connect()
  File "/srv/hinkskalle/backend/Hinkskalle/util/auth/ldap.py", line 36, in connect
    self.connection.rebind(user=self.bind_dn, password=self.bind_password)
  File "/usr/local/lib/python3.8/dist-packages/ldap3/core/connection.py", line 718, in rebind
    return self.bind(read_server_info, controls)
  File "/usr/local/lib/python3.8/dist-packages/ldap3/core/connection.py", line 589, in bind
    self.open(read_server_info=False)
  File "/usr/local/lib/python3.8/dist-packages/ldap3/strategy/sync.py", line 57, in open
    BaseStrategy.open(self, reset_usage, read_server_info)
  File "/usr/local/lib/python3.8/dist-packages/ldap3/strategy/base.py", line 154, in open
    raise LDAPSocketOpenError('invalid server address')
ldap3.core.exceptions.LDAPSocketOpenError: invalid server address
```
But login with my ldap user + pw works and the bind_dn and bind_pw was also checked.

A closer to the logs showed, that the `rq_worker` job does not read the values from `config.json`.This service seem to require everything in the env variables.

**Comment by grisuthedragon** [comment]
I found the solution. The `config.json` file needs to be added as volume to the `rq_worker` section,:

```
  rq_worker:
    image: ghcr.io/csf-ngs/hinkskalle:ldap-config-attr-mapping
    command: 'script/start-worker.sh'
    env_file:
      - "conf/secrets.env"
      - "conf/hinkskalle.env"
    volumes:
      - ./conf/config.json:/srv/hinkskalle/conf/config.json
    depends_on:
      - redis
      - hinkdb
```

**Comment by h3kker** [comment]
huh, of course. so far there wasn't really anything in the config.json for the workers, but now they need it

I've corrected the example deployment in 9e0c125. I also made the mount read-only for a little bit of added securtiy.

**Comment by grisuthedragon** [comment]
Now, this looks fine and works nicely.
